test/config/test_element.rb

```diff
@@ -63,4 +63,120 @@ def element(name = 'ROOT', arg = '', attrs = {}, elements = [])
       end
     end
   end
+
+  sub_test_case '#initialize' do
+    test 'creates object with blank attrs and elements' do
+      e = element('ROOT', '', {}, [])
+      assert_equal([], e.elements)
+    end
+
+    test 'creates object which contains attrs and elements' do
+      e = element('ROOT', '', {"k1" => "v1", "k2" => "v2"}, [
+                    element('test', 'mydata', {'k3' => 'v3'}, [])
+                  ])
+      assert_equal('ROOT', e.name)
+      assert_equal('',  e.arg)
+      assert_equal('v1', e["k1"])
+      assert_equal('v2', e["k2"])
+      assert_equal(1, e.elements.size)
+      e.each_element('test') do |e|
+        assert_equal('test', e.name)
+        assert_equal('mydata',  e.arg)
+        assert_equal('v3', e["k3"])
+      end
+    end
+  end
+
+  sub_test_case '#add_element' do
+    test 'elements can be set by #add_element' do
+      e = element()
+      assert_equal [], e.elements
+
+      e.add_element('e1', '')
+      e.add_element('e2', '')
+      assert_equal [element('e1', ''), element('e2', '')], e.elements
+    end
+  end
+
+  sub_test_case '#==' do
+    sub_test_case 'compare with two element objects' do
+      test 'equal' do
+        e1 = element('ROOT', '', {}, [])
+        e2 = element('ROOT', '', {}, [])
+        assert_true(e1 == e2)
+      end
+
+      test 'not equal' do
+        e1 = element('ROOT', '', {}, [])
+        e2 = element('ROOT', 'mydata', {}, [])
+        e3 = element('ROOT', 'mydata', {"k1" => "v1"}, [])
+        e4 = element('ROOT', 'mydata', {"k1" => "v1"}, [
+                       element('test', 'mydata', {'k3' => 'v3'}, [])
+                     ])
+        assert_false(e1 == e2)
+        assert_false(e2 == e3)
+        assert_false(e3 == e4)
+      end
+    end
+  end
+
+  sub_test_case '#+' do
```

Comment:
It's better to have tests like:
* overwriting existing attributes in e1 by e2
* adding elements in e2 to elements of e1